src/common/models/url-shortener/url-shortener.ts

```diff
@@ -29,7 +29,7 @@ export class UrlShortener implements Instance<UrlShortenerDef, UrlShortenerDef>
   public readonly shortenerFunction: UrlShortenerFn;
 
   constructor(private shortenerDefinition: string) {
-    this.shortenerFunction = new Function("url", "request", shortenerDefinition) as UrlShortenerFn;
+    this.shortenerFunction = new Function("url", "request", "options", shortenerDefinition) as UrlShortenerFn;
```

Comment:
I would like to think about this argument name. @mkuthan  feel free to give opinion. I was more thinking about “metadata”.

Reply:
I would introduce type, it could be ShortenerOptions, ShortenerMetadata or ShortenerContext (my vote for context because this is a placeholder for additional non-essential runtime context).

If it does not break backward compatibility, the argument could be orderer like this:
request, url, context (infrastructure, essential runtime context, additional runtime context)

Reply:
> I would introduce type, it could be ShortenerOptions, ShortenerMetadata or ShortenerContext (my vote for context because this is a placeholder for additional non-essential runtime context).

My vote goes for `context`



> If it does not break backward compatibility, the argument could be orderer like this:
> request, url, context (infrastructure, essential runtime context, additional runtime context)

Wouldn't break anything. Seems logical.

Reply:
Done (I like that order better too, and it's neat that it doesn't break anything to change it)